lib/operationQueryParameter.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { OperationParameterType } from "./operationParameterType";
+import { QueryCollectionFormat } from "./queryCollectionFormat";
+
+/**
+ * A parameter for an operation that will be added as a query parameter to the operation's HTTP
+ * request.
+ */
+export interface OperationQueryParameter {
+  /**
+   * The name of the parameter.
+   */
+  parameterName: string;
```

Comment:
what's the case where parameterName and queryParameterName varies? Does that come down to the Swagger?

Reply:
One common case I saw in the generator was `apiVersion` and `api-version`. There are also cases where the parameter name is `this.client.value`. I can probably optimize the later example so that it just turns into `value`, but for now I'm trying to keep it simple.